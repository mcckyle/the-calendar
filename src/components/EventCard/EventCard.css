/* Filename: EventCard.css */

/* ==================================
    Event Card Container Styling.
===================================== */
.event-card
{
    background: #fff;
    border-radius: var(--border-radius);

    padding: 1.25rem;
    margin: 0 auto;
    max-width: 600px;

    font-family: var(--body-font);
    color: var(--text-color);
    box-shadow: var(--shadow-sm);

    transition: box-shadow var(--transition-speed) ease,
                transform var(--transition-speed) ease;
}

.event-card:hover
{
    transform: translateY(-2px);
    box-shadow: var(--shadow-md);
}

/* =============================
    Header Styling.
================================ */
.event-card-header
{
    margin-bottom: 0.75rem;
}

.event-card-title
{
    font-size: 1.25rem;
    font-weight: 600;
    margin: 0 0 0.35rem;
    color: var(--deep-blue);
    line-height: 1.3;
}

/* ===============================
    Meta Information Styling.
================================== */
.event-card-meta
{
    display: flex;
    flex-wrap: wrap;
    gap: 0.4rem;

    font-size: 0.9rem;
    color: #6b7280; /* Color: Gray-500. */
}

.event-date
{
    font-weight: 500;
}

.event-time
{
    font-weight: 400;
}

/* ===============================
   Event Description Styling.
================================== */
.event-card-description
{
    font-size: 0.95rem;
    line-height: 1.55;
    color: #374151; /* Color: Gray-700. */
    margin-top: 0.75rem;
}

/* ===============================
   Event Venue Styling.
================================== */
.event-card-venue
{
    font-size: 0.9rem;
    color: #4b5563; /* Color: Gray-600. */
    margin-top: 0.85rem;
}

.venue-label
{
    margin-right: 0.3rem;
    color: var(--accent-gold);
}

/* ===============================
   Event Links Styling.
================================== */
.event-card-link a
{
    display: inline-block;
    margin-top: 1rem;
    font-size: 0.9rem;
    font-weight: 600;
    color: var(--accent-gold);
    text-decoration: none;

    position: relative;
    transition: color var(--transition-speed) ease;
}

.event-card-link a::after
{
    content: "";
    position: absolute;
    left: 0;
    bottom: -2px;
    height: 2px;
    width: 100%;
    background: currentColor;
    transform: scaleX(0);
    transform-origin: left;

    transition: transform var(--transition-speed) ease;
}

.event-card-link a:hover
{
    color: var(--deep-blue);
}

.event-card-link a:hover::after
{
    transform: scaleX(1);
}

/* ===============================
    Responsive Design Styling.
================================== */
@media (max-width: 768px)
{
    .event-card
    {
        padding: 1rem;
    }

    .event-card-title
    {
        font-size: 1.15rem;
    }
}

@media (max-width: 480px)
{
    .event-card
    {
        padding: 0.85rem;
    }

    .event-card-title
    {
        font-size: 1rem;
    }
}
