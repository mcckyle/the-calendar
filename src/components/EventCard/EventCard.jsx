import React from 'react';
import './EventCard.css';

const EventCard = ({ title, date, startTime, endTime, allDay, description }) => {
  // Check if startTime is a valid Date object and format it
  const formattedStartTime =
    startTime instanceof Date && !isNaN(startTime.getTime())
      ? startTime.toLocaleTimeString("en-US", {
          hour: "2-digit",
          minute: "2-digit",
        })
      : "No Start Time Available";

  // Format the full date using toLocaleDateString for CST time zone explicitly...
  //const formattedDate = new Date(date).toLocaleDateString('en-US', { weekday: 'long', month: 'long', day: 'numeric', year: 'numeric' });
	console.log("Raw Date Input:", date);

	const formattedDate = date
	  ? new Date(`${date}T00:00:00`).toLocaleDateString("en-US", {
		  weekday: "long",
		  month: "long",
		  day: "numeric",
		  year: "numeric",
		})
	  : "Invalid Date";

	return (
	  <div className="calendar-event" role="article" aria-labelledby={title}>
		<header className="event-header">
		  <h3 className="event-title">{title}</h3>
			  <p className="event-date-time">
				{formattedDate}{" "}
			  </p>
              <p>Start Time: {formattedStartTime}</p>

		</header>
		{description && <p className="event-description">{description}</p>}
	  </div>
	);
};

export default EventCard;
