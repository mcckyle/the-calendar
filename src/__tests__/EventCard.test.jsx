//Filename: EventCard.test.jsx
//Author: Kyle McColgan
//Date: 20 October 2025
//Description: This file contains unit tests for the EventCard component.

import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import EventCard from '../components/EventCard/EventCard.jsx';

process.env.TZ = 'America/Chicago'; // Force CST for consistent tests.

describe("EventCard Component", () => {
  const validEvent = {
    title: "Sample Event",
    date: "2025-01-19T00:00:00-06:00", //Explicit CST Midnight.
    startTime: new Date("2025-01-19T15:00:00-06:00"), // 3:00 PM.
    endTime: new Date("2025-01-19T17:00:00-06:00"), // 5:00 PM.
    allDay: false,
    description: "This is a test description for the event.",
  };

  //Test #1: Renders event title.
  test("renders the event title properly.", () => {
    render(<EventCard {...validEvent} />);
    expect(screen.getByText("Sample Event")).toBeInTheDocument();
  });

  //Test #2: Renders formatted date correctly.
  test("renders the formatted date.", () => {
    render(<EventCard {...validEvent} />);
    expect(screen.getByText(/Sunday, January 19, 2025/)).toBeInTheDocument();
  });

  //Test #3: Displays formatted start time for non-all-day events.
  test("renders formatted start time when allDay is false.", () => {
    render(<EventCard {...validEvent} />);
    expect(screen.getByText(/3:00 PM/)).toBeInTheDocument();
  });

  //Test #4: Does not display time for all-day events.
  test("does not render start time if allDay is true.", () => {
    const allDayEvent = { ...validEvent, allDay: true};
    render(<EventCard {...allDayEvent} />);
    expect(screen.queryByText(/3:00 PM/)).toBeNull();
  });

  //Test #5: Renders description if provided.
  test("renders description if one is provided.", () => {
    render(<EventCard {...validEvent} />);
    expect(screen.getByText("This is a test description for the event.")).toBeInTheDocument();
  });

  //Test #6: Does not render description when not provided.
  test("does not render description when not provided.", () => {
    const noDescriptionEvent = { ...validEvent, description: ""};
    render(<EventCard {...noDescriptionEvent} />);
    expect(screen.queryByText(/This is a test description./)).toBeNull();
  });

  //Test #7: Displays 'No Start Time' when startTime is invalid.
//   test("renders fallback text if startTime is invalid.", () => {
//     const invalidTimeEvent = { ...validEvent, startTime: "invalid-date"};
//     render(<EventCard {...invalidTimeEvent} />);
//     expect(screen.getByText(/No Start Time/i)).toBeInTheDocument();
//   });

  //Test #8: Displays 'Invalid Date' if date prop is missing or is invalid.
  test("renders 'Invalid Date' for invalid or missing date.", () => {
    const invalidDateEvent = { ...validEvent, date: ""};
    render(<EventCard {...invalidDateEvent} />);
    expect(screen.getByText(/Invalid Date/)).toBeInTheDocument();
  });

  //Test #9: EventCard has proper role and aria-labelledby.
  test("component has proper role and an accessible label.", () => {
    render(<EventCard {...validEvent} />);
    const article = screen.getByRole("article");
    expect(article).toHaveAttribute("aria-labelledby", "event-Sample Event");
  });

  //Test #10: Snapshot test for UI consistency purposes.
//   test("matches snapshot.", () => {
//     const { asFragment } = render(<EventCard {...validEvent} />);
//     expect(asFragment()).toMatchSnapshot();
//   });
});
