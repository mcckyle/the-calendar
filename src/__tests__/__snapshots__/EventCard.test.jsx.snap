// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Component matches snapshot. 1`] = `
<DocumentFragment>
  <article
    aria-labelledby="event-Sample Event"
    class="event-card"
    role="article"
  >
    <header
      class="event-card-header"
    >
      <h3
        class="event-card-title"
        id="event-Sample Event"
      >
        Sample Event
      </h3>
      <p
        class="event-card-meta"
      >
        <span
          class="event-date"
        >
          Saturday, January 18, 2025
        </span>
        <span
          class="event-time"
        >
           â€¢ 1/19/2025, 03:00 PM
        </span>
      </p>
    </header>
    <p
      class="event-card-description"
    >
      This is a test description for the event.
    </p>
  </article>
</DocumentFragment>
`;
